### PR TITLE
Niftany

### PR DESCRIPTION
**Pull request by awead**
Applies changes to update Niftany (and Rubocop) to their latest versions.

@rschenk there's some funny business going on with Rubocop's uniqueness validators.

### CODE REVIEW

**Comment by awead** [comment]
@rschenk Rubocop is failing the uniqueness check on AddIndexOnAccessLevelToAccessControls but as far as I can tell, it seems right to me.

**Comment by rschenk** [comment]
I did some digging into Rubocop, and I think this is actually a bug/oversight in Rubocop.

Check out this output, which I inserted into Rubocop's validator thingy:
```
$ bundle exec rubocop app/models/access_control.rb
Inspecting 1 file
================================Rubocop-Index=================================
  Rubocop's expected column names: 
    #<Set: {"access_level", "agent_id", "resource_id"}>
  Indexed columns:
    #<Set: {"access_level", "agent_type", "resource_type", "agent_id", "resource_id"}>
------------------------------------------------------------------------------
```

Because these are polymorphic relationships, it's not enough to _just_ index the `*_id`. For example, let's just look at Agents, supposing you had....
agent_type | agent_id
----------- | ---------
Group | 123
User | 123

If you only index by agent_id, this would fail the uniqueness constraint, which is improper.

tldr; you are right and rubocop is wrong. So there.
